Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -1625,27 +1628,74 @@ static bool isContextMenuEvent(const WebMouseEvent& event)
-    menuItems.append({ WebCore::contextMenuItemPDFTwoPages(), currentDisplayMode == ContextMenuItemTag::TwoPages, enumToUnderlyingType(ContextMenuItemTag::TwoPages), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No });
-    menuItems.append({ WebCore::contextMenuItemPDFTwoPagesContinuous(), currentDisplayMode == ContextMenuItemTag::TwoPagesContinuous, enumToUnderlyingType(ContextMenuItemTag::TwoPagesContinuous), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No });
+    menuItems.appendVector(displayModeContextMenuItems());
 
     addSeparator();
 
-    menuItems.append({ WebCore::contextMenuItemPDFZoomIn(), 0, enumToUnderlyingType(ContextMenuItemTag::ZoomIn), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No });
-    menuItems.append({ WebCore::contextMenuItemPDFZoomOut(), 0, enumToUnderlyingType(ContextMenuItemTag::ZoomOut), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No });
-    menuItems.append({ WebCore::contextMenuItemPDFActualSize(), 0, enumToUnderlyingType(ContextMenuItemTag::ActualSize), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No });
+    menuItems.appendVector(scaleContextMenuItems());
 
     return { contextMenuPoint, WTFMove(menuItems), { enumToUnderlyingType(ContextMenuItemTag::OpenWithPreview) } };
 }
 
+PDFContextMenuItem UnifiedPDFPlugin::separatorContextMenuItem() const
+{
+    return { { }, 0, enumToUnderlyingType(ContextMenuItemTag::Invalid), ContextMenuItemEnablement::Disabled, ContextMenuItemHasAction::No, ContextMenuItemIsSeparator::Yes };
+}
+
+Vector<PDFContextMenuItem> UnifiedPDFPlugin::selectionContextMenuItems(const IntPoint& contextMenuPoint) const
+{
+    if (![m_pdfDocument allowsCopying] || !m_currentSelection)
+        return { };
+
+    Vector<PDFContextMenuItem> items {
+        { contextMenuItemTagSearchWeb(), 0, enumToUnderlyingType(ContextMenuItemTag::WebSearch), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No },
+        separatorContextMenuItem(),
+        { contextMenuItemTagLookUpInDictionary(selectionString()), 0, enumToUnderlyingType(ContextMenuItemTag::DictionaryLookup), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No },
+        separatorContextMenuItem(),
+        { contextMenuItemTagCopy(), 0, enumToUnderlyingType(ContextMenuItemTag::Copy), ContextMenuItemEnablement::Enabled, ContextMenuItemHasAction::Yes, ContextMenuItemIsSeparator::No },
+    };
+
+    if (pdfElementTypesForPluginPoint(contextMenuPoint).contains(PDFElementType::Link))
```

Comment:
`contextMenuPoint` is not in the right coordinate space. Investigating.

Reply:
Fixed in https://github.com/WebKit/WebKit/commit/063ec2d2a085a96d3f3c25231e56c43d00ff69a6